src/services/vehicleDragger.ts

```diff
@@ -165,10 +214,23 @@ function updateVehicleDrag(args: DragVehicleArgs): void
 		return;
 	}
 
-	const position = args.position;
-	car.x = position.x;
-	car.y = position.y;
-	car.z = position.z;
+	if (isNull(args.position.trackPosition)) {
+		const position = args.position.tilePosition;
+		car.x = position.x;
+		car.y = position.y;
+		car.z = position.z;
+	} else {
+		car.trackLocation = args.position.trackPosition;
+		if (!isNull(args.position.trackProgress)) {
+			car.travelBy(getDistanceFromProgress(car, args.position.trackProgress));
+		} else {
+			// internally, travelBy calls MoveTo so the car will render on the holding
+			// track
+			// HACK: seems to make the vehicle update render properly
```

Comment:
Please no hacks here.

Reply:
I think we need a new scripting API for it. something like `Car.redraw()`

Reply:
Hmm it sounds more like setting the track location property should just invalidate/redraw the vehicle. Such a fix could also be done/merged separately without having to put this PR in a wait for a new API. 🙂

Reply:
Umm I still think the hack is needed though. for now at least. Having the vehicle render in the wrong place can be unpleasent to use.

Reply:
Would it be possible to just `travelBy` once without the `getDestination` method?
```ts
car.travelBy(1);
```
Or possibly with 0 instead of 1 if that works too.